python/lsst/obs/decam/decamMapper.py

```diff
@@ -73,11 +89,18 @@ def __init__(self, inputPolicy=None, **kwargs):
                                      DecamMapper._nbit_filter)
 
     def _extractDetectorName(self, dataId):
+        copyId = self._transformId(dataId)
         try:
-            return DecamMapper.detectorNames[dataId['ccdnum']]
+            return DecamMapper.detectorNames[copyId['ccdnum']]
         except KeyError:
             raise RuntimeError("No name found for dataId: %s"%(dataId))
 
+    def _transformId(self, dataId):
+        copyId = dataId.copy()
+        if "ccd" in copyId:
+            copyId.setdefault("ccdnum", copyId["ccd"])
```

Comment:
Under what circumstances would `ccdnum` exist but not be equal to `ccd` -- should that be an error?

Why are we copying the dataId here? It's not immediately obvious why we can't update it in place. In fact, I think the name of the method—"transform"—implies that this will modify its argument, not return a copy.